### PR TITLE
add `command list --device <devicename>

### PR DESCRIPTION
**Pull request by difince**
add `command list --device <devicename>
--device is mandaroty field

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

File: cmd/command/list/list.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package list
+
+import (
+	"github.com/spf13/viper"
```
Comment (mkbhanda) [inline cmd/command/list/list.go:17]: should this import be at the very end?

File: cmd/command/list/list.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+func listHandler(cmd *cobra.Command, args []string) (err error) {
+	url := config.Conf.Clients["Command"].Url() + clients.ApiDeviceRoute + "/name/" + device
+	var response models.CommandResponse
+	err = request.Get(url, &response)
+	if err != nil {
+		return
+	}
+	pw := viper.Get("writer").(io.WriteCloser)
+	w := new(tabwriter.Writer)
+	w.Init(pw, 0, 8, 1, '\t', 0)
```
Comment (mkbhanda) [inline cmd/command/list/list.go:59]: Would you add a comment to explain what all these magic numbers are 0, 8, 1

File: cmd/command/list/list.go (hunk 1 above)
Comment (difince) [reply cmd/command/list/list.go:17]: It should be :) Thank you

File: cmd/command/list/list.go (hunk 2 above)
Comment (difince) [reply cmd/command/list/list.go:66]: These numbers are described in `w.Init`method itself.

> // A Writer must be initialized with a call to Init. The first parameter (output)
// specifies the filter output. The remaining parameters control the formatting:
//
//	**minwidth**	minimal cell width including any padding
//	**tabwidth**	width of tab characters (equivalent number of spaces)
//	**padding**		padding added to a cell before computing its width
//	**padchar**		ASCII char used for padding
//			        if padchar == '\t', the Writer will assume that the
//	         		width of a '\t' in the formatted output is tabwidth,
//		        	and cells are left-aligned independent of align_left
//			       (for correct-looking results, tabwidth must correspond
//			       to the tab width in the viewer displaying the result)
//	**flags**		formatting control
//
func (b *Writer) Init(output io.Writer, minwidth, tabwidth, padding int, padchar byte, flags uint) *Writer {

So, I think there is not need to duplicate this information in all the places that uses `Writer.Init` method.

**Comment by difince** [comment]
@mkbhanda I have added support to get All  Commands , but without --limit flag. This is because `edgex-go` does not support the functionality you have requested - " 50 internal maximum". Probably, this list could get pretty exhausting, but this endpoint still exists in edgex-go, so I have added it.

File: cmd/command/list/list.go (hunk 2 above)
Comment (difince) [reply cmd/command/list/list.go:66]: In follow-up PR, I plan to extract the logic with these magical numbers in a common place used from all the commands, so this place could be appropriate for adding this clarification.

**Pull request merged**